src/FlowDiagnostics.jl

```diff
@@ -1,35 +1,45 @@
 module FlowDiagnostics
 
+export RichardsonNumber, RossbyNumber
+export ErtelPotentialVorticityᶠᶠᶠ, ThermalWindPotentialVorticityᶠᶠᶠ
+
 using Oceananigans.Operators
+using Oceananigans.AbstractOperations
+using Oceananigans.AbstractOperations: KernelFunctionOperation
 using KernelAbstractions: @index, @kernel
 using Oceananigans.Grids: Center, Face
 using Oceananigans.Fields: KernelComputedField
 
 # Some useful operators
 @inline fψ²(i, j, k, grid, f, ψ) = @inbounds f(i, j, k, grid, ψ)^2
 
-@kernel function richardson_number_ccf!(Ri, grid, u, v, b, params)
-    i, j, k = @index(Global, NTuple)
 
-    dBdz_tot = ∂zᵃᵃᶠ(i, j, k, grid, b)        + params.N2_bg   # dbdz(c, c, f)
-    dUdz_tot = ℑxᶜᵃᵃ(i, j, k, grid, ∂zᵃᵃᶠ, u) + params.dUdz_bg # dudz(f, c, f) → dudz(c, c, f)
-    dVdz_tot = ℑyᵃᶜᵃ(i, j, k, grid, ∂zᵃᵃᶠ, v) + params.dVdz_bg # dvdz(c, f, f) → dvdz(c, c, f)
+function RichardsonNumber(model; N²_bg=0, dUdz_bg=0, dVdz_bg=0)
+    u, v, w = model.velocities
+    b = model.tracers.b
+
+    dBdz_tot = ∂z(b) + N²_bg
+    dUdz_tot = ∂z(u) + dUdz_bg
+    dVdz_tot = ∂z(v) + dVdz_bg
 
-    @inbounds Ri[i, j, k] = dBdz_tot / (dUdz_tot^2 + dVdz_tot^2)
+    return dBdz_tot / (dUdz_tot^2 + dVdz_tot^2)
 end
 
 
-@kernel function rossby_number_ffc!(Ro, grid, u, v, params)
-    i, j, k = @index(Global, NTuple)
+function RossbyNumber(model; dUdy_bg=0, dVdx_bg=0)
+    u, v, w = model.velocities
+    f₀ = model.coriolis.f
```

Comment:
This only works with `FPlane`. Is that intended?

Alternatively this object could take `u, v, f` as arguments rather than `model`; then users can provide correct values as needed, and because we're using abstract operations it might even work with something like

```
f(x, y, z) = f0 + beta * y
```

since we can use functions in AbstractOperations.

Reply:
Yeah it only works with f-plane. This was only because I didn't think to pass functions to `AbstractOperations`!

But I think your idea is good and it appears to work (I just tested it). The caveat is that this makes it more likely that this won't compile in GPUs depending on the function

Reply:
GPU seems to be able to compile at least reasonable functions, so  adopted your suggestion